include/aws/crt/http/HttpConnection.h

```diff
@@ -356,6 +371,8 @@ namespace Aws
                  * not be freed until the stream is completed.
                  *
                  * Returns an instance of HttpStream upon success and nullptr on failure.
+                 *
+                 * You must call HttpClientStream::Activate() to begin outgoing processing of the stream.
                  */
                 std::shared_ptr<HttpClientStream> NewClientStream(const HttpRequestOptions &requestOptions) noexcept;
```

Comment:
Add a unit test that proves we clean up properly if the stream is never activated.
I know python did tricks to keep it alive from creation->on_complete and this had to change to only do tricks from activate->on_complete